tests/unit/test_graphstates.py

```diff
@@ -111,8 +111,22 @@ def test_SCZ_mat(self, random_graph):
             assert np.array_equal(mat[N:, :N], random_graph[1])
             assert np.array_equal(mat[N:, N:], np.identity(N))
 
-    # def test_SCZ_apply(self):
-    # pass
+    @pytest.mark.parametrize("one_shot", [True, False])
+    def test_SCZ_apply(self, random_graph, one_shot):
+        """test SCZ application with `one_shot` parameter"""
+        delta = rng().random()
+        model_init = noise_model(delta, "initial")
+        model_fin = noise_model(delta, "final")
+
+        n = len(random_graph[0])
+        G = CVLayer(random_graph[0])
+        G.apply_noise(model_init)
+        init_noise_all_GKP = np.full(2 * n, (delta / 2) ** 0.5, dtype=np.float32)
+        assert np.array_equal(G._init_noise, init_noise_all_GKP)
+
+        G.apply_noise(model_fin)
+        noise_cov_all_GKP = SCZ_apply(G._adj, np.diag(init_noise_all_GKP) ** 2, one_shot=one_shot)
+        assert np.array_equal(G._noise_cov.toarray(), noise_cov_all_GKP)
 
```

Comment:
This is good, but it has a lot of moving parts. In addition to this test, I recommend a test independent of CVLayer. e.g. it could just apply SCZ to some vector and check that it produces what is expected; or it can test some properties (like multiplying to the identity, or if it has the structure of: identity + adjacency matrix on the bottom-left corner). We don't necessarily have to do it in this PR. If we don't, I would suggest to make a very low priority shortcut ticket out of it.

Reply:
I have created a ticket for it. [SC-18500]